std/socket.zig

```diff
@@ -0,0 +1,678 @@
+pub usingnamespace @import("errors.zig");
+usingnamespace @import("errors.zig");
+
+const std = @import("../index.zig");
+const builtin = @import("builtin");
+const net = std.net;
+const os = std.os;
+
+const is_windows = builtin.os == Os.windows;
+
+const is_posix = switch (builtin.os) {
+    builtin.Os.linux, builtin.Os.macosx, builtin.Os.freebsd => true,
+    else => false,
+};
+
+const unexpectedError = switch (builtin.os) {
+    builtin.Os.windows => system.unexpectedError,
+    else => os.unexpectedErrno,
+};
+
+pub const SocketFd = switch (builtin.os) {
+    builtin.Os.windows => system.SOCKET,
+    else => i32,
+};
+
+pub const InvalidSocketFd: SocketFd = switch (builtin.os) {
+    builtin.Os.windows => system.INVALID_SOCKET,
+    else => -1,
+};
+
+pub const Domain = enum {
```

Comment:
Never seen `Domain` before.  This is usually called `AddressFamily`.  This type is the "tag" of the address union for sockets.